### PR TITLE
[Fix] Added Components dropdown functionality, resolved small devices dropdown issues

### LINKED ISSUE #5694: Component of Sistent should have dropdown
### Description
A component section of sistent does not have dropdown within the table of content.

### Expected Behavior
It should have dropdown like identity in table of content.

### Screenshots

> 
Screenshot from 2024-07-19 09-59-16

### Environment:
- Host OS: Ubuntu
- Browser: Chrome

---
<h2>Contributor Resources and <a href="https://layer5.io/community/handbook">Handbook</a></h2>

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- 📚 See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md).
- 🎨 Wireframes and [designs for Layer5 site](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](https://slack.layer5.io).

Join the Layer5 Community by submitting your [community member form](https://layer5.io/newcomer).

### PR DESCRIPTION
**Pull request by Muhammed-Moinuddin**
**Description**
Made Table of Content (button) and dropdown workable on small devices.
Added dropdown functionality to components.

This PR fixes #5694 #5677 

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit c0059d0ea216425d33d26f01e6b12c5f58668f13 at: https://66a0d372d54a9cf62f93852f--layer5.netlify.app

File: src/components/SistentNavigation/index.js [hunk 1]
```diff
@@ -114,13 +117,70 @@ const TOC = () => {
...
+                      className={`toc-sub-heading toc-sub-inline element-item ${
+                        location.pathname.includes(
+                          "/projects/sistent/components/button"
+                        )
+                          ? "active"
+                          : ""
+                      }`}
+                      activeClassName="active"
+                    >
+                      Button
```
Comment (sudhanshutech) [inline src/components/SistentNavigation/index.js:148]: why they are static? Shouldn't they be dynamic

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: src/components/SistentNavigation/index.js (hunk 1 above)
Comment (Muhammed-Moinuddin) [reply src/components/SistentNavigation/index.js:148]: Just followed the previous way of code (i.e. static). Dynamic would be great. I'll look into it.

Comment (sudhanshutech) [reply src/components/SistentNavigation/index.js:148]: thanks, please propose a change if you would

Comment (Muhammed-Moinuddin) [reply src/components/SistentNavigation/index.js:148]: Yes, I will.

**Comment by Ashparshp** [comment]
@Muhammed-Moinuddin Thanks for your contribution, let's discuss this on the website's call. Please add this as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk).

**Comment by Muhammed-Moinuddin** [comment]
> @Muhammed-Moinuddin Thanks for your contribution, let's discuss this on the website's call. Please add this as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk).

Not fruitful. I need to make changes in it.

**Comment by l5io** [comment]
🚀 Preview for commit be5170d18d6d41a4ba111eed3eae8a84c84d7f8a at: https://66b8ccf5a3dcf17322b4843e--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit f2cbf60c6135d01be085c0d5c5f3eef79d198cd4 at: https://66b915c1e24d819efd07976a--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 56ac93d4c275cbc726286fb7e1f06e6d25242dca at: https://66b924e1fdff58b4813d4342--layer5.netlify.app

**Comment by Ashparshp** [comment]
@Muhammed-Moinuddin, Thank you! Let's discuss this during the website call and get it merged.  Please consider adding it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk) || [meeting link](https://meet.layer5.io/websites).

**Comment by sudhanshutech** [comment]
@Muhammed-Moinuddin please resolve the merge conflicts. so we can move ahead.

**Comment by Ashparshp** [comment]
@Muhammed-Moinuddin //

**Comment by sudhanshutech** [comment]
@Muhammed-Moinuddin checking in again , solving the conflicts?

**Comment by Muhammed-Moinuddin** [comment]
> src/sections/Meshmap/features/features.style.js

I tried, but there are too many conflicts. I'll re-raise the PR

**Comment by sudhanshutech** [comment]
@Muhammed-Moinuddin you still here?
There are only 2 conflicts

**Comment by sudhanshutech** [comment]
@Muhammed-Moinuddin checking in again?

**Comment by sudhanshutech** [comment]
@Muhammed-Moinuddin

**Comment by sudhanshutech** [comment]
I am closing this pr since its been abandoned.

**Pull request closed without merging**